crates/network-scanner/src/interfaces/mod.rs

```diff
@@ -0,0 +1,27 @@
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "windows")]
+mod windows;
+
+#[cfg(target_os = "windows")]
+pub use windows::get_network_interfaces;
+
+#[cfg(target_os = "linux")]
+pub use linux::get_network_interfaces;
+
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NetworkInterface {
+    name: String,
+    description: Option<String>,
+    mac_address: Vec<[u8; 6]>,
```

Comment:
yes it is, but not common, one example is [MAC address spoofing](https://en.wikipedia.org/wiki/MAC_spoofing)

Reply:
Correct me if I'm wrong, but if you spoof your MAC address, you are "changing" the MAC address of the network interface. You are not adding an additional MAC address. As a result, you still only have a single MAC address, not two. Of course, a single device can have many MAC addresses. One per network interface (be it virtual or physical). But each network interface doesn't have more than one MAC address at a time, even if it can be changed at will. Can you double check this?

Reply:
you are right, one network interface at the same time can only have one mac address, will update.

Reply:
updated